src/pyodide/upload_bundles.py

```diff
@@ -0,0 +1,90 @@
+import json
+import subprocess
+import sys
+from copy import deepcopy
+from functools import cache
+from os import environ
+from pathlib import Path
+
+import requests
+from boto3 import client
+
+
+def cquery(rule):
+    res = subprocess.run(
+        [
+            "bazel",
+            "cquery",
+            rule,
+            "--output=files",
+        ],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    if res.returncode:
+        print(res.stdout)
+        print(res.stderr)
+        sys.exit(res.returncode)
+    return res.stdout.strip()
+
+
+@cache
+def _bundle_version_info():
+    with Path(cquery("@workerd//src/pyodide:bundle_version_info")).open() as f:
+        return json.load(f)
+
+
+def bundle_version_info():
+    return deepcopy(_bundle_version_info())
+
+
+def get_pyodide_bin_path(ver):
+    return cquery(f"@workerd//src/pyodide:pyodide@{ver}.capnp")
+
+
+def bundle_key(*, pyodide_version, pyodide_date, backport, **_kwds):
+    return f"pyodide_{pyodide_version}_{pyodide_date}_{backport}.capnp.bin"
+
+
+def bundle_url(**kwds):
+    return "https://pyodide-capnp-bin.edgeworker.net/" + bundle_key(**kwds)
+
+
+def get_backport(ver):
+    info = bundle_version_info()[ver]
+    backport = int(info["backport"])
+    for b in range(backport + 1, backport + 10):
+        info["backport"] = b
+        url = bundle_url(**info)
+        res = requests.head(url)
+        if res.status_code == 404:
+            return b
+        res.raise_for_status()
+
+
+def main():
+    s3 = client(
+        "s3",
+        endpoint_url=f"https://{environ['R2_ACCOUNT_ID']}.r2.cloudflarestorage.com",
+        aws_access_key_id=environ["R2_ACCESS_KEY_ID"],
+        aws_secret_access_key=environ["R2_SECRET_ACCESS_KEY"],
+        region_name="auto",
+    )
+
+    for ver, info in bundle_version_info().items():
+        if ver.startswith("dev"):
+            continue
+        path = Path(get_pyodide_bin_path(ver)).resolve()
+        print("path", path)
+        print(Path.cwd())
+        b = get_backport(ver)
+        info["backport"] = b
+        key = bundle_key(**info)
+        print(f"Uploading version {ver} backport {b}")
+        s3.upload_file(str(path), "pyodide-capnp-bin", key)
```

Comment:
I assume upload_file will print some stuff to stdout, so maybe add another print here to show what version and backport was uploaded so that it's easy to find by just going to the end of the execution output.

Reply:
`upload_file()` is completely silent so this isn't necessary.